utils/drift_detection.py

```diff
@@ -94,3 +91,31 @@ def detect_uncertainty_drift(dataset, train_loader, model):
 
                 if preds['data']['is_drift']:       # removing drifted samples from buffer
                     model.buffer.flush_class(cls)
+
+def detect_uncertainty_label_drift(dataset, train_loader, model):
+    drifting_classes = dataset.drifting_classes
+    if len(drifting_classes) == 0:
+        return
+
+    labels = ['No!', 'Yes!']
+
+    for cls in drifting_classes:
+        filtered_images = []
+        for img_batch, _, _, original_target_batch in train_loader:
+            mask = original_target_batch == cls
+            selected_images = img_batch[mask]
+            filtered_images.append(selected_images)
+
+        new_images = torch.cat(filtered_images, dim=0)
+
+    if new_images.size(0) > 0 and hasattr(model, 'buffer'):
```

Comment:
it looks like this if should be inside for loop, but in Github it look like its not.

Reply:
It is actually inside the for loop in my VS Code, I don't know why it's appearing outside here. 
![Screenshot 2024-08-15 134330](https://github.com/user-attachments/assets/e7706efb-061b-4d27-8cfe-2bd81e790b7b)

Reply:
You are right; I changed it during this implementation. I was thinking of collecting all the fine-grained classes and then comparing them with the metaclasses from the buffer since we don't know which metaclass comes from which fine-grained class. However, if we incorporate the fine-grained labels inside the buffer, this problem will be fixed.